src/main/java/android/pages/CartPage.java

```diff
@@ -0,0 +1,50 @@
+package android.pages;
+
+import com.zebrunner.carina.utils.factory.DeviceType;
+import com.zebrunner.carina.webdriver.decorator.ExtendedWebElement;
+import com.zebrunner.carina.webdriver.decorator.PageOpeningStrategy;
+import common.pages.CartPageBase;
+import common.pages.ProductsPageBase;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.support.FindBy;
+
+import java.util.List;
+
+@DeviceType(pageType = DeviceType.Type.ANDROID_PHONE, parentClass = CartPageBase.class)
+public class CartPage extends CartPageBase {
+
+    @FindBy(xpath = "//android.widget.TextView[@text='YOUR CART']")
+    private ExtendedWebElement title;
+
+    @FindBy(xpath = "//android.view.ViewGroup[@content-desc='test-Item']")
+    private List<ExtendedWebElement> products;
+
+    @FindBy(xpath = "//android.widget.TextView[@text='CONTINUE SHOPPING']")
+    private ExtendedWebElement continueShoppingButton;
+
+    @FindBy(xpath = "//android.view.ViewGroup[@content-desc='test-CHECKOUT']")
+    private ExtendedWebElement checkOutButton;
+
+    public CartPage(WebDriver driver) {
+        super(driver);
+        setPageOpeningStrategy(PageOpeningStrategy.BY_ELEMENT);
+        setUiLoadedMarker(title);
+    }
+
+    @Override
+    public int getProductsSize() {
+        return products.size();
+    }
+
+    @Override
+    public ProductsPageBase clickContinueShoppingButton() {
+        continueShoppingButton.click();
+        return initPage(getDriver(), ProductsPageBase.class);
```

Comment:
`getDriver()` can be removed in all `initPage`, it'll get default current driver
```suggestion
        return initPage(ProductsPageBase.class);
```

Reply:
fixed